Modules/Core/Common/include/itkBSplineInterpolationWeightFunction.hxx

```diff
@@ -18,14 +18,19 @@
 #ifndef itkBSplineInterpolationWeightFunction_hxx
 #define itkBSplineInterpolationWeightFunction_hxx
 
-#include "itkBSplineInterpolationWeightFunction.h"
 #include "itkBSplineKernelFunction.h"
 #include "itkImage.h"
 #include "itkMatrix.h"
 #include "itkMath.h"
 
 namespace itk
 {
+
+// Definition of static constexpr must be outside of class until C++17, see .h file for declaration & initialization
```

Comment:
Can you please add something like `#if __cplusplus < 201703L` here? With C++17 it is actually deprecated to define a static constexpr outside of the class definition!!!

In the [Working Draft of the C++ Standard](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/n4901.pdf) it says:
> For compatibility with prior C++ International Standards, a `constexpr` static data member may be redundantly
redeclared outside the class with no initializer. This usage is deprecated.